src/clock.sh

```diff
@@ -1,22 +1,57 @@
 #!/bin/bash
 
 function clock::now() {
-  if perl -MTime::HiRes -e "" > /dev/null 2>&1; then
-    perl -MTime::HiRes -e 'printf("%.0f\n", Time::HiRes::time() * 1000)'
-  elif [[ "${_OS:-}" != "OSX" ]]; then
+  if dependencies::has_perl && perl -MTime::HiRes -e "" > /dev/null 2>&1; then
+    if perl -MTime::HiRes -e 'printf("%.0f\n",Time::HiRes::time()*1000000000)'; then
+        return 0
+    fi
+  fi
+
+  if  ! check_os::is_macos && ! check_os::is_alpine; then
```

Comment:
What about ubuntu? I think ubuntu is also pretty close to alpine in the sense that they are both linux, wouldn't make sense to check if it's not macos neither ubuntu? Also, I am wondering what about instead of checking negative cases, what about checking a positive condition like `is_windows` only?